### PR TITLE
add `scotto` community layouts and userspace

### PR DESCRIPTION
**Pull request by dunk2k**
## Description
- community layouts
    - split_3x5_3
    - split_3x5_2
    - ortho_4x10
- userspace
    - scotto

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by dunk2k** [comment]
@joe-scotto : placed this PR into draft until it gets greenlight from yourself

**Comment by joe-scotto** [comment]
@dunk2k Looks good, but I want to change some things... can I get push access to the repo?

**Comment by dunk2k** [comment]
@joe-scotto 
Added you as collaborator to my forked repo - hopefully that gives you required permissions :)

**Comment by joe-scotto** [comment]
@dunk2k That worked, I was able to also test all 4 maps and we should be good to open the PR now. 

There are 4 maps that basically are all the same but it adds distinction for QWERTY or Colemak users. There might be a better way to name these though so open to suggestions.

**LAYOUT_split_3x5_2**
- scotto-qwerty
- scotto-colemak

**LAYOUT_split_3x5_3**
- scotto-qwerty
- scotto-colemak

**Comment by dunk2k** [comment]
@joe-scotto 
> There are 4 maps that basically are all the same but it adds distinction for QWERTY or Colemak users. There might be a better way to name these though so open to suggestions.

I've added both scotto layouts to ortho_4x10 layout - up to you whether it's worth keeping.

I'm happy to promote this PR from draft status after your word on ortho_4x10.

Possible improvements:
- move `config.h` in each `keymap.c`, finger keys in keymap, and tap dance macros to a `scotto` userspace (`users/scotto`) 
- program a means so that:
    1. `scotto-colemak` and `scotto-qwerty` keymaps dirs are merged into `scotto` dir,
    2. when inputting, in QMK CLI, `-km scotto-qwerty`, the *scotto-qwerty* keymap is loaded. Same premise for `scotto-colemak`

**Comment by joe-scotto** [comment]
@dunk2k 

I think that's perfect as long as we keep in mind that the TD functions are slightly different from `3x5_2` and `3x5_3` but I guess that shouldn't really matter since all the names are different between them.

**Comment by dunk2k** [comment]
@joe-scotto 
> I think that's perfect as long as we keep in mind that the TD functions are slightly different from 3x5_2 and 3x5_3  

Thanks for pointing that out! I'll be sure work that into future improvements 😁

P.S. moving PR out of draft status

**Comment by dunk2k** [comment]
QMK Collaborators:
Forgetting the `scotto-qwerty` folder(s) at this stage, could someone steer me in right direction as to why build failures keep occuring with the following error messages?
```
Linking: .build/a_dux_scotto.elf                                                                    [ERRORS]
 |
 | c:/program files/qmk_msys/mingw64/bin/../lib/gcc/avr/8.5.0/../../../../avr/bin/ld.exe: .build/obj_a_dux_scotto/quantum/process_keycode/process_tap_dance.o: in function `preprocess_tap_dance':
 | C:\Users\duncan.sutherland\GitHub\qmk_firmware/quantum/process_keycode/process_tap_dance.c:126: undefined reference to `tap_dance_actions'
 | c:/program files/qmk_msys/mingw64/bin/../lib/gcc/avr/8.5.0/../../../../avr/bin/ld.exe: C:\Users\duncan.sutherland\GitHub\qmk_firmware/quantum/process_keycode/process_tap_dance.c:126: undefined reference to `tap_dance_actions'
 | c:/program files/qmk_msys/mingw64/bin/../lib/gcc/avr/8.5.0/../../../../avr/bin/ld.exe: C:\Users\duncan.sutherland\GitHub\qmk_firmware/quantum/process_keycode/process_tap_dance.c:127: undefined reference to `tap_dance_actions'
 | c:/program files/qmk_msys/mingw64/bin/../lib/gcc/avr/8.5.0/../../../../avr/bin/ld.exe: C:\Users\duncan.sutherland\GitHub\qmk_firmware/quantum/process_keycode/process_tap_dance.c:127: undefined reference to `tap_dance_actions'
 | c:/program files/qmk_msys/mingw64/bin/../lib/gcc/avr/8.5.0/../../../../avr/bin/ld.exe: .build/obj_a_dux_scotto/quantum/process_keycode/process_tap_dance.o: in function `process_tap_dance':
 | C:\Users\duncan.sutherland\GitHub\qmk_firmware/quantum/process_keycode/process_tap_dance.c:146: undefined reference to `tap_dance_actions'
 | c:/program files/qmk_msys/mingw64/bin/../lib/gcc/avr/8.5.0/../../../../avr/bin/ld.exe: .build/obj_a_dux_scotto/quantum/process_keycode/process_tap_dance.o:C:\Users\duncan.sutherland\GitHub\qmk_firmware/quantum/process_keycode/process_tap_dance.c:146: more undefined references to `tap_dance_actions' follow
 | collect2.exe: error: ld returned 1 exit status
 |
make[1]: *** [builddefs/common_rules.mk:268: .build/a_dux_scotto.elf] Error 1
```

Is it related to these following lines in `tap_dance.c`?:
```
void x_finished(tap_dance_state_t *state, void *user_data);
void x_reset(tap_dance_state_t *state, void *user_data);
```

**Comment by tzarc** [comment]
You haven't added `tap_dance.c` to the `SRC +=` line in the userspace `rules.mk`.

**Comment by dunk2k** [comment]
> You haven't added `tap_dance.c` to the `SRC +=` line in the userspace `rules.mk`.
@tzarc Sincerely thank you!

**Comment by dunk2k** [comment]
Hi @joe-scotto ,

Improvements completed! would you be able to test/give your blessing?

The `readme.md` in `users/scotto` directory shows command to compile the QWERTY version of your personal (functional) layout.

Once all your keyboard related PRs are merged, you've the option to remove the `scotto` keymap from your keyboards whose layout macro has been assigned as either:
- LAYOUT_ortho_4x10
- LAYOUT_split_3x5_2
- LAYOUT_split_3x5_3

File: users/scotto/config.h
Comment (joe-scotto) [inline users/scotto/config.h:21]:
```suggestion
        #define TAPPING_TERM 135
```

I use 135 in my keymaps, should change this.

File: users/scotto/readme.md
Comment (joe-scotto) [inline users/scotto/readme.md:14]:
```suggestion
Example for [Scotto34](/keyboards/scottokeebs/scotto34):
```
Might be able to change this to a ScottoKeeb

**Review by joe-scotto — CHANGES_REQUESTED** [review]
A few small changes, however when testing I'm unable to use the map:

```
❯ qmk flash -kb scottokeebs/scotto34 -km scotto -e KEYMAP=QWERTY
☒ Invalid keymap argument.
usage: qmk flash [-h] [-c] [-e ENV] [-j PARALLEL] [-n] [-km KEYMAP] [-kb KEYBOARD] [-m MCU] [-bl BOOTLOADER] [-b] [filename]
```

**Comment by dunk2k** [comment]
@joe-scotto 

As always with troubleshotting, back to basics checks:
1. Have you pulled latest version of this branch to your testing machine?
2. Is the `scotto34` keyboard's layout macro set to `LAYOUT_split_3x5_2`?
3.  Is there a `scotto` keymap inside `scotto34` keyboard directory that *maybe* is conflicting?

I'm thinking more like number 3, as files in the `scotto34/keymaps/scotto` directory doesn't have accommodations for any keymap argument(s). I've successfully compiled this userspace keymap for `crkbd`, `a_dux`, and `keebformom` keyboards.

**Comment by joe-scotto** [comment]
@dunk2k Yeah, I'm on the latest, layout is correct, and there is no scotto keymap in that folder.

**Comment by dunk2k** [comment]
When you say 'error', does QMK show an error or is it that flashed keyboard doesn't functioning as intended?

**Comment by joe-scotto** [comment]
@dunk2k The error is just `Invalid keymap argument.`

**Comment by dunk2k** [comment]
@joe-scotto 
I've copied over a scotto34 keyboard and am getting same error. I'll keep at it and let you know the resolution :)

**Comment by dunk2k** [comment]
wait a minute!
scotto34 is missing, in `info.json` file,  `"community_layouts": ["split_3x5_2"],`

**Comment by joe-scotto** [comment]
@dunk2k That worked! So yeah, everything looks good then. What we should do is try to get this merged and then once all my current PRs are merged I can open a big PR to just update all of their info.json files to include the layout property.

**Comment by dunk2k** [comment]
@joe-scotto 

I and many countless others have PRs awaiting to be approved. I'm a humble, yet keen, user just like yourself, so I've honestly no idea what the structure/process is internally at QMK once official collaborators have given their approval - which doesn't help at all considering you're running a self-employed business using this firmware.

P.S. I've added comments on 'Scotto 34 PCB' PR to help modernize it and place config/definitions at proper level 😉

**Review by joe-scotto — APPROVED** [review]

**Comment by dunk2k** [comment]
@joe-scotto thanks 😊

File: layouts/community/ortho_4x10/scotto/keymap.c
Comment (drashna) [inline layouts/community/ortho_4x10/scotto/keymap.c:16]: Since "scotto.h" has the qmk keyboard h included, you don't need to include it here.
```suggestion
#include "scotto.h"
```

File: users/scotto/config.h
Comment (drashna) [inline users/scotto/config.h:25]: Just a nit.
```suggestion
#ifdef TAP_DANCE_ENABLE
#    ifdef TAPPING_TERM
#        undef TAPPING_TERM
#        define TAPPING_TERM 135
#    endif
#    define TAPPING_TERM_PER_KEY
#    define PERMISSIVE_HOLD
#endif
```

File: users/scotto/rules.mk
Comment (drashna) [inline users/scotto/rules.mk:10]:
```suggestion
    SRC += tap_dance.c
endif

ifneq ($(strip $(KEYMAP)),)
    OPT_DEFS += -DKEYMAP_$(KEYMAP)
```

File: users/scotto/tap_dance.c
Comment (drashna) [inline users/scotto/tap_dance.c:154]:
```suggestion
        case LT(2, KC_ENT) :
            return 200;
        default:
            return TAPPING_TERM;
    }
};
```

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**
